src/test/types.ts

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2016 Palantir Technologies, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+export const FILE_EXTENSION = ".linttest";
```

Comment:
the nice thing about it is that it's not _exactly_ a test -- it's a source file with extra linting error markup or a description of expected linter output. so, yes, it is primarily useful for generating tests, but theoretically it doesn't have to be limited to that. let's do `.ts.lint` then?